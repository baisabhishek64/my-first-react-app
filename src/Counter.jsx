import React,{Component} from 'react';


export default class Counter extends Component{
    
    constructor(){
        super(...arguments);
        this.state = {
            counter: 0
        }
        this.incrementCount = this.incrementCount.bind(this);
    }

    incrementCount(){
        this.setState({
            counter: this.state.counter+1
        });
    }

    render(){
        return(
            <div>
                {this.state.counter}
                <br />
                <button type="button" onClick={this.incrementCount}>Increase</button>
            </div>
        )
    }

}