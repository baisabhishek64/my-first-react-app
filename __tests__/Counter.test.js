import React from 'react';
import {shallow} from 'enzyme';
import Counter from '../src/Counter';

describe('Counter suite tests', () => {
    it('Counter rendered properly', () => {
            const component = shallow(<Counter />);
            expect(component.getElements()).toMatchSnapshot();
    });
});