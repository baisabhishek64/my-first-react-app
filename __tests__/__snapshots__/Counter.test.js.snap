// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter suite tests Counter rendered properly 1`] = `
Array [
  <div>
    0
    <br />
    <button
      onClick={[Function]}
      type="button"
    >
      Increase
    </button>
  </div>,
]
`;
